### DISCUSSION TITLE
Is there any kind of Python API?

### DISCUSSION THREAD

**Discussion by venlx** [openthread/openthread#9004] [Q&A] [answered]
I have a Python Script where I use the Openthread CLI on my Border Router (Raspberry Pi with a nRF52840 Dongle). So far I had no problems executing the commands using subprocess. Now I want to read received messages (currently UDP messages) but I don't find a good way to do that using my current method. So I wanted to ask whether there is a Python API or general ideas how I could solve the problem?

**Comment by jwhui** [comment, marked as answer]
Thread transports IP packets and appears as a network interface. You can use the [sockets](https://docs.python.org/3/library/socket.html) interface to send/receive UDP messages.
